### ISSUE TITLE
SIGSEGV (Address boundary error) when build.zig changed

### ISSUE THREAD

**Issue by MFAshby** [ziglang/zig#19427]
### Zig Version

0.12.0-dev.3433+640acf862

### Steps to Reproduce and Observed Behavior

- `zig init` to make a new project
- `zig build` build it first time
- edit build.zig file, any trivial edit will do e.g. add a comment
- `zig build` again
- SIGSEGV signal is received.

```
martin@martin-pinebook ~/d/zig-issue> zig init
info: created build.zig
info: created build.zig.zon
info: created src/main.zig
info: created src/root.zig
info: see `zig build --help` for a menu of options
martin@martin-pinebook ~/d/zig-issue> zig build
martin@martin-pinebook ~/d/zig-issue> echo "// some comment" >> build.zig
martin@martin-pinebook ~/d/zig-issue> zig build
fish: Job 1, 'zig build' terminated by signal SIGSEGV (Address boundary error)
martin@martin-pinebook ~/d/zig-issue [SIGSEGV]>
```

Note I am on linux-aarch64
```
martin@martin-pinebook ~/d/zig-issue> uname -a
Linux martin-pinebook 6.6.7-1-MANJARO-ARM #1 SMP PREEMPT Sun Dec 17 15:40:27 UTC 2023 aarch64 GNU/Linux
```

### Expected Behavior

The project should successfully compile and run

**Comment by MFAshby** [comment]
If I remove the `zig-cache` folder and try again, it builds successfully. Not sure why but maybe it's a clue.

**Comment by Rexicon226** [comment]
was fixed here: af0668d6c2c7696e695cf0553c1ae2d593e9effe

**Comment by squeek502** [comment]
Duplicate of https://github.com/ziglang/zig/issues/19408, fixed in https://github.com/ziglang/zig/pull/19417
